api/tests/certification/enrolment/unit/domain/models/UserEligibilityCalculator_test.js

```diff
@@ -0,0 +1,234 @@
+        });
+      });
+
+      it('should compute core certificable as false when user is level 1 on less than 5 competences', function () {
+        // given
+        const someDate = new Date('2021-10-29');
+        // competence E
+        allKnowledgeElements.push(
+          domainBuilder.buildKnowledgeElement({
+            competenceId: 'compE',
+            earnedPix: 2,
+          }),
+        );
+        const userEligibilityCalculator = domainBuilder.certification.enrolment.buildUserEligibilityCalculator({
+          userId: 123,
+          date: someDate,
+          eligibilities: [],
+          eligibilitiesV2: [],
+        });
+
+        // when
+        userEligibilityCalculator.computeCoreEligibility({ allKnowledgeElements, coreCompetences });
+
+        // then
+        expect(userEligibilityCalculator.toDTO()).to.deep.equal({
+          userId: 123,
+          date: someDate,
+          eligibilities: [{ certification: LABEL_FOR_CORE, isCertifiable: false }],
+          eligibilitiesV2: [{ certification: LABEL_FOR_CORE, isCertifiable: false }],
+        });
+      });
+
+      it('should compute core certifiable as false when user is level 1 on less than 5 competences (no KE in one competence)', function () {
+        // given
+        const someDate = new Date('2021-10-29');
+        const userEligibilityCalculator = domainBuilder.certification.enrolment.buildUserEligibilityCalculator({
+          userId: 123,
+          date: someDate,
+          eligibilities: [],
+          eligibilitiesV2: [],
+        });
+
+        // when
+        userEligibilityCalculator.computeCoreEligibility({ allKnowledgeElements, coreCompetences });
+
+        // then
+        expect(userEligibilityCalculator.toDTO()).to.deep.equal({
+          userId: 123,
+          date: someDate,
+          eligibilities: [{ certification: LABEL_FOR_CORE, isCertifiable: false }],
+          eligibilitiesV2: [{ certification: LABEL_FOR_CORE, isCertifiable: false }],
+        });
+      });
+    });
+  });
+
+  context('#buildUserEligibilityList', function () {
+    context('when computing is not done', function () {
+      it('should throw an Error stating that computing is not done', function () {
+        // given
+        const someDate = new Date('2020-01-01');
+        const userEligibilityCalculator = domainBuilder.certification.enrolment.buildUserEligibilityCalculator({
+          userId: 123,
+          date: someDate,
+          eligibilities: [],
+          eligibilitiesV2: [],
+        });
+
+        // when
+        const error = catchErrSync(userEligibilityCalculator.buildUserEligibilityList, userEligibilityCalculator)();
+
+        // then
+        expect(error).to.be.instanceof(Error);
+        expect(error.message).to.equal('Cannot produce final UserEligibilityList before computing them.');
+      });
+
+      it('should return a UserEligibilityList because computing is done', function () {
```

Comment:
**remarque** : le it de succès ici se trouve à l'intérieur du context de l'error (when computing is not done), il faudrait le sortir de là bichette (pour pas reprendre ton expression)